subsys/fs/littlefs_fs.c

```diff
@@ -37,79 +37,35 @@ static K_MEM_SLAB_DEFINE(file_data_pool, sizeof(struct lfs_file_data),
 static K_MEM_SLAB_DEFINE(lfs_dir_pool, sizeof(struct lfs_dir),
 			 CONFIG_FS_LITTLEFS_NUM_DIRS, 4);
 
-/* If either filecache memory pool is customized by either the legacy
- * mem_pool or heap Kconfig options then we need to use a heap.
- * Otherwise we can use a fixed region.
- */
-#if defined(CONFIG_FS_LITTLEFS_FC_MEM_POOL)		\
-	|| ((CONFIG_FS_LITTLEFS_FC_HEAP_SIZE - 0) > 0)
-#define FC_ON_HEAP 1
-#else
-#define FC_ON_HEAP 0
-#endif
-
-#if FC_ON_HEAP
-
 /* Inferred overhead, in bytes, for each k_heap_aligned allocation for
  * the filecache heap.  This relates to the CHUNK_UNIT parameter in
  * the heap implementation, but that value is not visible outside the
  * kernel.
  */
 #define FC_HEAP_PER_ALLOC_OVERHEAD 24U
 
-/* If not explicitly customizing provide a default that's appropriate
- * based on other configuration options.
- */
-#ifndef CONFIG_FS_LITTLEFS_FC_MEM_POOL
-#define CONFIG_FS_LITTLEFS_FC_MEM_POOL_MAX_SIZE CONFIG_FS_LITTLEFS_CACHE_SIZE
-#define CONFIG_FS_LITTLEFS_FC_MEM_POOL_NUM_BLOCKS CONFIG_FS_LITTLEFS_NUM_FILES
-#endif
-
-/* If not explicitly customizing infer a default from the legacy
- * mem-pool configuration options.
- */
 #if (CONFIG_FS_LITTLEFS_FC_HEAP_SIZE - 0) <= 0
+/* Autho-generate heap size from cache size and number of files */
```

Comment:
was this suppose to be auto-generate?

Reply:
Yes. The Kconfig comment for the `CONFIG_FS_LITTLEFS_FC_HEAP_SIZE` states that if <=0 it will be picked for user.

Reply:
I guess the question was is this a typo? `Autho` v `Auto`

Reply:
Yep. You got me there. Fixing.